### PR TITLE
Feature/tvist1 233 merge changes

### PR DESCRIPTION
**Pull request by jekuaitk**
* Added basic information edit template and logic

### CODE REVIEW

File: src/Controller/CaseController.php [hunk 1]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
      */
-    public function information(CaseEntity $case): Response
+    public function information(CaseEntity $case, ComplaintCategoryRepository $categoryRepository): Response
     {
+        /** @var ComplaintCategory $complaintCategory */
+        $complaintCategory = $categoryRepository->findOneBy([
+            'board' => $case->getBoard(),
+            'municipality' => $case->getMunicipality(),
+            'name' => $case->getCaseType(),
+        ]);
```
Comment (lats1) [inline src/Controller/CaseController.php:52]: ComplaintCatetory seems to be case specific, so it should probably be a property on the CaseEntity class.

File: src/Controller/CaseController.php [hunk 2]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
     /**
      * @Route("/{id}/information", name="case_information", methods={"GET"})
      */
-    public function information(CaseEntity $case): Response
+    public function information(CaseEntity $case, ComplaintCategoryRepository $categoryRepository): Response
```
Comment (lats1) [inline src/Controller/CaseController.php:45]: This controller method shows the CaseEntity's data, and should follow the convention in other Symfony projects. Hence this method should be renamed to `public function show` and the route should be `@Route("/{id}")` and the route name should be `case_show`

File: src/Controller/CaseController.php [hunk 3]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
         return $this->render('case/information.html.twig', [
             'case' => $case,
+            'complaintCategory' => $complaintCategory,
+        ]);
+    }
+
+    /**
+     * @Route("/{id}/information/edit", name="case_information_edit", methods={"GET", "POST"})
+     */
+    public function editInformation(CaseEntity $case, ComplaintCategoryRepository $categoryRepository, Request $request): Response
```
Comment (lats1) [inline src/Controller/CaseController.php:63]: Same as above with the showInformation method. This method should be renamed to `edit`and the route should be `@Route("/{id}/edit)" and the route name should be `case_edit`.

File: src/Controller/CaseController.php [hunk 4]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+    {
+        /** @var ComplaintCategory $complaintCategory */
+        $complaintCategory = $categoryRepository->findOneBy([
+            'board' => $case->getBoard(),
+            'municipality' => $case->getMunicipality(),
+            'name' => $case->getCaseType(),
+        ]);
```
Comment (lats1) [inline src/Controller/CaseController.php:70]: Same as above where complaintCatogory should be a property on the CaseEntity class.

File: src/Form/ResidentComplaintBoardCaseType.php [hunk 5]
```diff
@@ -41,11 +41,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('complainantPostalCode')
             ->add('caseType', ChoiceType::class, [
                 'choices' => [
-                    $caseTypesAssociative,
+                    'Complaint Categories' => $caseTypesAssociative,
```
Comment (lats1) [inline src/Form/ResidentComplaintBoardCaseType.php:44]: Should come directly from the case entity object.

File: templates/case/information.html.twig [hunk 6]
```diff
@@ -31,7 +31,7 @@
         <div class="col-auto">
             {# TODO: Needs functionality on buttons #}
             <a name="" id="" class="btn btn-success" href="#" role="button">{% trans %}Create representative{% endtrans %}</a>
-            <a name="" id="" class="btn btn-primary" href="#" role="button">{% trans %}Edit basic information{% endtrans %}</a>
+            <a name="" id="" class="btn btn-primary" href="{{ url('case_information_edit', {id: case.id}) }}" role="button">{% trans %}Edit basic information{% endtrans %}</a>
```
Comment (lats1) [inline templates/case/information.html.twig:34]: The value for the route name parameter in the url function should be `case_edit`.

File: src/Controller/CaseController.php [hunk 7]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+        if ($form->isSubmitted() && $form->isValid()) {
+            $case = $form->getData();
+
+            $em = $this->getDoctrine()->getManager();
+            $em->flush();
+
+            return $this->redirectToRoute('case_information', [
+                'id' => $case->getId(),
+                'case' => $case,
+                'complaintCategory' => $complaintCategory,
```
Comment (lats1) [inline src/Controller/CaseController.php:90]: This is probably not needed as complaint category should be a property on the case entity class.

File: src/Controller/CaseController.php [hunk 8]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+            ]);
+        }
+
+        return $this->render('case/information_edit.html.twig', [
+            'case' => $case,
+            'complaintCategory' => $complaintCategory,
```
Comment (lats1) [inline src/Controller/CaseController.php:96]: This is probably not needed as complaint category should be a property on the case entity class.

File: src/Controller/CaseController.php [hunk 9]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+    public function information(CaseEntity $case, ComplaintCategoryRepository $categoryRepository): Response
     {
+        /** @var ComplaintCategory $complaintCategory */
+        $complaintCategory = $categoryRepository->findOneBy([
+            'board' => $case->getBoard(),
+            'municipality' => $case->getMunicipality(),
+            'name' => $case->getCaseType(),
+        ]);
+
         return $this->render('case/information.html.twig', [
```
Comment (lats1) [inline src/Controller/CaseController.php:54]: Also the related template should be renamed `case/show.twig.html`

File: src/Controller/CaseController.php [hunk 10]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+        $form = $this->createForm(ResidentComplaintBoardCaseType::class, $case, ['board' => $case->getBoard()]);
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $case = $form->getData();
+
+            $em = $this->getDoctrine()->getManager();
+            $em->flush();
+
+            return $this->redirectToRoute('case_information', [
```
Comment (lats1) [inline src/Controller/CaseController.php:87]: The route should be `case_show`.

File: src/Controller/CaseController.php [hunk 11]
```diff
@@ -38,10 +42,60 @@ public function summary(CaseEntity $case): Response
...
+                'id' => $case->getId(),
+                'case' => $case,
+                'complaintCategory' => $complaintCategory,
+            ]);
+        }
+
+        return $this->render('case/information_edit.html.twig', [
```
Comment (lats1) [inline src/Controller/CaseController.php:94]: The template should be renamed `case/edit.html.twig`.

**Review by lats1 — CHANGES_REQUESTED** [review]

File: src/Controller/CaseController.php (hunk 1 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:52]: Added complaintCategory property on the CaseEntity class.

File: src/Controller/CaseController.php (hunk 4 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:70]: Added complaintCategory property on the CaseEntity class.

File: src/Controller/CaseController.php (hunk 7 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:90]: Removed as so longer necessary per addition of complaintCategory property on CaseEntity.

File: src/Controller/CaseController.php (hunk 8 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:96]: Removed as so longer necessary per addition of complaintCategory property on CaseEntity.

File: src/Form/ResidentComplaintBoardCaseType.php (hunk 5 above)
Comment (jekuaitk) [reply src/Form/ResidentComplaintBoardCaseType.php:44]: That would not give the user any choice. But it needs modification after complaintCategory property addition on CaseEntity. 

Switched to EntityType and got 'choices' from Board.

File: src/Controller/CaseController.php (hunk 2 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:45]: Renamed route and function accordingly.

File: src/Controller/CaseController.php (hunk 3 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:63]: Renamed route and function accordingly.

File: templates/case/information.html.twig (hunk 6 above)
Comment (jekuaitk) [reply templates/case/information.html.twig:34]: Used renamed route name `case_edit.`

File: src/Controller/CaseController.php (hunk 9 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:54]: Used renamed template `case/show.html.twig`.

File: src/Controller/CaseController.php (hunk 10 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:87]: Used renamed route name.

File: src/Controller/CaseController.php (hunk 11 above)
Comment (jekuaitk) [reply src/Controller/CaseController.php:94]: Used renamed template name `case/edit.html.twig`.

File: src/Controller/SidebarController.php [hunk 12]
```diff
@@ -62,7 +62,7 @@ public function renderSubmenu(UuidV4 $caseId, string $activeRoute): Response
     {
         $submenuItems = [
             $this->generateSubmenuItem('Summary', 'case_summary', $caseId, $activeRoute),
-            $this->generateSubmenuItem('Basic information', 'case_information', $caseId, $activeRoute),
+            $this->generateSubmenuItem('Basic information', 'case_show', $caseId, $activeRoute),
```
Comment (lats1) [inline src/Controller/SidebarController.php:65]: Good catch!

File: src/Entity/ComplaintCategory.php [hunk 13]
```diff
@@ -2,15 +2,19 @@
...
 use Doctrine\ORM\Mapping as ORM;
 use Symfony\Bridge\Doctrine\IdGenerator\UuidV4Generator;
 use Symfony\Component\Uid\UuidV4;
 
 /**
  * @ORM\Entity(repositoryClass=ComplaintCategoryRepository::class)
+ * @ORM\EntityListeners({"App\Logging\EntityListener\ComplaintCategoryListener"})
  */
-class ComplaintCategory
+class ComplaintCategory implements LoggableEntityInterface
```
Comment (lats1) [inline src/Entity/ComplaintCategory.php:17]: Good catch!

**Review by lats1 — APPROVED** [review]

**Pull request merged**